tensorflow/python/eager/def_function.py

```diff
@@ -417,6 +417,19 @@ def embedding_matmul(a, b):
     self._input_signature = input_signature
     self._call_counter = _CallCounter(FREQUENT_TRACING_WARNING_MAX_CALL_HISTORY)
 
+  def __getstate__(self):
+    """Custom pickling, to omit unpickleable objects."""
+    result = self.__dict__.copy()
+    del result["_lock"]
```

Comment:
We should also make sure the lock is released to prevent deadlocks.

Reply:
This is a copy of the object. It doesn't affect the original object in any way. And when it's unpickled it creates new, unlocked lock. So it should be fine.